packages/typescript/src/components/TypeFunction.tsx

```diff
@@ -0,0 +1,89 @@
+import {
+  childrenArray,
+  findKeyedChild,
+  Prose,
+  Scope,
+  Show,
+} from "@alloy-js/core";
+import { Children } from "@alloy-js/core/jsx-runtime";
+import { getCallSignatureProps } from "../utils.js";
+import { CallSignature, CallSignatureProps } from "./CallSignature.jsx";
+import { Declaration } from "./Declaration.jsx";
+import {
+  FunctionParameters,
+  FunctionTypeParameters,
+  getReturnType,
+  TypeFunctionParameters,
+} from "./FunctionBase.jsx";
+import { JSDoc } from "./JSDoc.jsx";
+import { JSDocParams } from "./JSDocParam.jsx";
+
+/** Define a TypeScript type function
+ *
+ * @example `(foo: string, bar: number) => void`
+ */
+export interface TypeFunctionProps extends CallSignatureProps {
+  /** If the method is async */
+  readonly async?: boolean;
+  /** Documentation for this method. */
+  readonly doc?: Children;
+  readonly children?: Children;
+}
+
+/**
+ * A TypeScript function declaration.
+ *
+ * @remarks
+ *
+ * Providing parameters and type parameters can be accomplished in one of three
+ * ways:
+ *
+ * 1. As an array of {@link TypeParameterDescriptor}s or
+ *    {@link TypeParameterDescriptor}s.
+ * 2. As raw content via the `parametersChildren` or `typeParametersChildren`
+ *    props.
+ * 3. As a child of this component via the
+ *    {@link (TypeFunction:namespace).Parameters} or
+ *    {@link (TypeFunction:namespace).TypeParameters} components.
+ */
+export function TypeFunction(props: TypeFunctionProps) {
```

Comment:
I feel like `FunctionType` is more correct, it's the name given in the TypeScript grammar I believe.